### PR TITLE
JSON to CSV

### PR DESCRIPTION
**Pull request by gvela024**
Started, still need to do the temperature

### CODE REVIEW

**Comment by JohnLe01** [comment]
FAIL!? 😭

File: src/diagnosticComponents.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  'IR Receiver',
+  'Upper Switch',
+  'Lower Switch',
+  'UV LED',
+  'LED Light',
+  'LED Ring',
+  'LED Daytime',
+  'Clean Switch',
+  'Bucket Switch',
+];
```
Comment (JohnLe01) [inline src/diagnosticComponents.js]: 👍  I like it!

File: src/Translator.js [hunk 2]
```diff
@@ -1,19 +1,4 @@
...
-  'Upper Switch',
-  'Lower Switch',
-  'UV LED',
-  'LED Light',
-  'LED Ring',
-  'LED Daytime',
-  'Clean Switch',
-  'Bucket Switch',
-]
+var components = require('diagnosticComponents');
```
Comment (JohnLe01) [inline src/Translator.js]: all 'requires' are usually at the top of the file.

File: src/diagnosticComponents.js (hunk 1 above)
Comment (gvela024) [reply src/diagnosticComponents.js]: Didn't work and I didn't want to spend the time figuring it out. :/

Comment (JohnLe01) [reply src/diagnosticComponents.js]: :(

**Comment by JohnLe01** [comment]
👍

File: src/Translator.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ var components = [
   'LED Daytime',
   'Clean Switch',
   'Bucket Switch',
-]
+];
```
Comment (JohnLe01) [inline src/Translator.js:16]: lol 😁

File: spec/Translator_spec.js [hunk 4]
```diff
@@ -12,78 +12,76 @@ describe('Translator', function() {
     var actual = translator.translateToObject(diagnosticsData);
 
     var expected = {
-      diagnostics: {
```
Comment (gvela024) [inline spec/Translator_spec.js:15]: I actually had a bug in here. It was getting all the same data for each component.

File: src/Translator.js [hunk 5]
```diff
@@ -13,40 +13,31 @@ var components = [
...
-          diagnostics: {}
-        },
-        temporaryEntry: {}
-      };
+      var diagnostics = {}
 
-      for (var diagnosticEntryIndex = 0; diagnosticEntryIndex < Object.keys(diagnosticsData.rawDiagnostics).length; diagnosticEntryIndex++) {
-        context.entry = diagnosticsData.rawDiagnostics[diagnosticEntryIndex];
-        components.forEach(component, context);
-        context.result.diagnostics[diagnosticEntryIndex] = context.temporaryEntry;
```
Comment (gvela024) [inline src/Translator.js:31]: Note: assigning one object to another in JS just points to the other object, it doesn't copy the contents.

**Comment by gvela024** [comment]
This is ready to go.

File: spec/Translator_spec.js (hunk 4 above)
Comment (JohnLe01) [reply spec/Translator_spec.js:15]: Hm.. strange 😲

File: src/Translator.js (hunk 5 above)
Comment (JohnLe01) [reply src/Translator.js:31]: Hm.. interesting. Didn't know that fully

**Comment by JohnLe01** [comment]
You wizard 😆

**Pull request merged**